Tests/FluentAssertions.Specs/Primitives/EnumAssertionSpecs.cs

```diff
@@ -697,5 +698,98 @@ public void When_nullable_enum_is_not_null_it_should_throw()
         }
 
         #endregion
+
+        #region Match
+
+        [Fact]
+        public void Matching_the_predicate_should_not_throw()
+        {
+            // Arrange
+            BindingFlags flags = BindingFlags.Public;
+
+            // Act / Assert
+            flags.Should().Match(x => x == BindingFlags.Public);
+        }
+
+        [Fact]
+        public void Not_matching_the_predicate_should_throw_with_the_predicate_in_the_message()
+        {
+            // Arrange
+            BindingFlags flags = BindingFlags.Public;
+
+            // Act
+            Action act = () => flags.Should().Match(x => x == BindingFlags.Static, "that's what we need");
+
+            // Assert
+            act.Should().Throw<XunitException>().WithMessage("Expected*Static*because that's what we need*found*Public*");
+        }
+
+        [Fact]
+        public void A_null_predicate_should_not_be_accepted()
+        {
+            // Act
+            Action act = () => BindingFlags.Public.Should().Match(null);
+
+            // Assert
+            act.Should().Throw<ArgumentNullException>().WithMessage("*null*predicate*");
+        }
+
+        #endregion
+
+        #region Be One Of
+
+        [Fact]
+        public void Does_not_throw_if_it_is_one_of_the_expected_enums()
+        {
+            // Arrange
+            BindingFlags flags = BindingFlags.Public;
+
+            // Act / Assert
+            flags.Should().BeOneOf(BindingFlags.Public, BindingFlags.ExactBinding);
+        }
+
+        [Fact]
+        public void Throws_when_the_value_is_not_one_of_the_expected_enums()
+        {
+            // Arrange
+            BindingFlags flags = BindingFlags.DeclaredOnly;
+
+            // Act / Assert
+            Action act = () => flags.Should().BeOneOf(new[] { BindingFlags.Public, BindingFlags.ExactBinding }, "that's what we need");
+
+            act.Should()
+                .Throw<XunitException>()
+                .WithMessage("Expected*Public*ExactBinding*because that's what we need*found*DeclaredOnly*");
+        }
+
+        [Fact]
+        public void Cannot_be_one_of_an_empty_list_of_enums()
```

Comment:
🔧 It would be nice to form a real sentence by including a subject.

Some suggestions:
* `An_enum_cannot_be_one_of_an_empty_list`
* `An_enum_cannot_be_part_of_an_empty_list`
* `An_enum_is_not_part_of_an_empty_list`
* `An_empty_list_cannot_contain_an_enum`
* `An_empty_list_does_not_contain_an_enum`